azurelinuxagent/common/protocol/wire.py

```diff
@@ -768,6 +796,35 @@ def _update_from_goal_state(self, refresh_type):
 
         raise ProtocolError("Exceeded max retry updating goal state")
 
+    def _save_goal_state_if_necessary(self, new_goal_state, refresh_type):
+        # We cache the goal state if one of the following is true
+        # 1) We didn't previously have a goal state
+        # 2) The incarnation changed
+        # 3) The extensions config changed
+        # 4) The extensions config did not change, but previously it did
+        # 5) We have an update type of GoalStateForced
+        save_goal_state = False
+        if new_goal_state is None:
```

Comment:
it seems like new_goal_state can never be None

Reply:
I believe you're correct based on looking at the code. Removed.